xdp-forward/xdp-forward.c

```diff
@@ -258,6 +297,24 @@ static int do_unload(const void *cfg, __unused const char *pin_root_path)
 		pr_info("Unloaded from interface %s\n", iface->ifname);
 	}
 
+	snprintf(pin_link_path, sizeof(pin_link_path), "%s/%s", pin_root_path,
+			 "xdp_check_return");
+
+	link = bpf_link__open(pin_link_path);
```

Comment:
Instead of going through this open/unpin/destroy dance, just unlink the pin file?